xds/internal/client/xds.go

```diff
@@ -278,6 +281,64 @@ func processServerSideListener(lis *v3listenerpb.Listener) (*ListenerUpdate, err
 	return lu, nil
 }
 
+func validateNetworkFilterChains(filterChains []*v3listenerpb.FilterChain) error {
+	for _, filterChain := range filterChains {
+		if filterChain == nil {
+			continue
```

Comment:
Why would there be a `nil` message in a `repeated FilterChain` element?  I'm not even sure this is possible using the proto APIs in any language.

Reply:
I didn't want to special case for the `default_filter_chain` at the call site. So, I decided to add this nil check here. If you see around line 272, I make the following single call:

```golang
	if err := validateNetworkFilterChains(append(lis.GetFilterChains(), lis.GetDefaultFilterChain())); err != nil {
		return nil, err
	}
```

Even if I take two arguments for this function: one a slice of filter_chains, and one a default_filter chain, i would have to do a nil check for the latter.

What do you think?

Reply:
In that case, how about appending only if it's non-nil?
```go
chains := lis.GetFilterChains()
if def := lis.GetDefaultFilterChain(); def != nil {
  chains = append(chains, def)
}
if err := validateNetworkFilterChains(chains); err != nil {
```

Reply:
Done.